[iOS] Date picker toolbars may be clipped or sized inconsistently

lilyspiniolas opened the pull request:
#### 87634cab839d1909153f6a551f71ff6ad774c063
<pre>
[iOS] Date picker toolbars may be clipped or sized inconsistently
<a href="https://bugs.webkit.org/show_bug.cgi?id=293028">https://bugs.webkit.org/show_bug.cgi?id=293028</a>
<a href="https://rdar.apple.com/151356798">rdar://151356798</a>

Reviewed by Aditya Keerthi.

The toolbar within date pickers may be sized inconsistently
or clipped depending on the location of the popover arrow.
Adjust the toolbar&apos;s layout constraints when necessary to
correct this.

* Source/WebKit/UIProcess/ios/forms/WKDatePickerPopoverController.h:
* Source/WebKit/UIProcess/ios/forms/WKDatePickerPopoverController.mm:
(-[WKDatePickerPopoverView adjustLayoutIfNeeded]):
(-[WKDatePickerPopoverView setupView:toolbarBottomMargin:]):
(-[WKDatePickerPopoverView bottomMarginForToolbar]):
(-[WKDatePickerPopoverView toolbarTrailingConstraint]):
(-[WKDatePickerPopoverView toolbarLeadingConstraint]):
(-[WKDatePickerPopoverView toolbarBottomConstraint]):
(-[WKDatePickerPopoverView toolbarHeightConstraint]):
(-[WKDatePickerPopoverController viewSafeAreaInsetsDidChange]):
(adjustedMarginIfNeeded): Deleted.

Canonical link: <a href="https://commits.webkit.org/294978@main">https://commits.webkit.org/294978@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8088ff823c15900386b064cfc7a6329bdb45fd6d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/103688 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23390 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/13710 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/108878 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/54340 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/23734 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/31936 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/108878 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/54340 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/106694 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/18412 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/93537 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/108878 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/18220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/53716 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/88003 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/11644 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111267 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/30844 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/22743 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/87782 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31205 "Failed to checkout and rebase branch from PR 45420") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/89738 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/87432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/10039 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/25201 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16829 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/30772 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/36075 "Built successfully") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/30566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/33901 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/32127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/ios/forms/WKDatePickerPopoverController.h:
```diff
@@ -44,6 +44,7 @@
 - (void)presentInView:(UIView *)view sourceRect:(CGRect)rect completion:(void(^)())completion;
 - (void)dismissDatePicker;
 - (void)assertAccessoryViewCanBeHitTestedForTesting;
+- (void) viewSafeAreaInsetsDidChange;
```
No need to declare this. `UIViewController.h` has the declaration.

pxlcoder commented on Source/WebKit/UIProcess/ios/forms/WKDatePickerPopoverController.mm:
```diff
@@ -225,6 +261,14 @@ - (void)resetDatePicker
+- (void)viewSafeAreaInsetsDidChange
+{
+}
+#endif
```
Can be simplified.
```suggestion
#if USE(APPLE_INTERNAL_SDK) && __has_include(<WebKitAdditions/WKDatePickerPopoverControllerAdditions.mm>)
#import <WebKitAdditions/WKDatePickerPopoverControllerAdditions.mm>
#endif
```

The default implementation of `viewSafeAreaInsetsDidChange` will be used.

lilyspiniolas committed:
[iOS] Date picker toolbars may be clipped or sized inconsistently
https://bugs.webkit.org/show_bug.cgi?id=293028
rdar://151356798

Reviewed by Aditya Keerthi.

The toolbar within date pickers may be sized inconsistently
or clipped depending on the location of the popover arrow.
Adjust the toolbar's layout constraints when necessary to
correct this.

* Source/WebKit/UIProcess/ios/forms/WKDatePickerPopoverController.h:
* Source/WebKit/UIProcess/ios/forms/WKDatePickerPopoverController.mm:
(-[WKDatePickerPopoverView adjustLayoutIfNeeded]):
(-[WKDatePickerPopoverView setupView:toolbarBottomMargin:]):
(-[WKDatePickerPopoverView bottomMarginForToolbar]):
(-[WKDatePickerPopoverView toolbarTrailingConstraint]):
(-[WKDatePickerPopoverView toolbarLeadingConstraint]):
(-[WKDatePickerPopoverView toolbarBottomConstraint]):
(-[WKDatePickerPopoverView toolbarHeightConstraint]):
(-[WKDatePickerPopoverController viewSafeAreaInsetsDidChange]):
(adjustedMarginIfNeeded): Deleted.

Canonical link: https://commits.webkit.org/294978@main
